pkg/cloud/cloud.go

```diff
@@ -233,6 +260,38 @@ func (c *cloud) DescribeAccessPoint(ctx context.Context, accessPointId string) (
 	}, nil
 }
 
+func (c *cloud) findAccessPointByPath(ctx context.Context, volumeName string, accessPointOpts *AccessPointOptions) (accessPoint *AccessPoint, err error) {
+	klog.V(5).Infof("List AP input : %+v", accessPointOpts)
+	describeAPInput := &efs.DescribeAccessPointsInput{
+		FileSystemId: &accessPointOpts.FileSystemId,
+	}
+	res, err := c.efs.DescribeAccessPointsWithContext(ctx, describeAPInput)
+	if err != nil {
+		if isAccessDenied(err) {
+			return
+		}
+		if isFileSystemNotFound(err) {
+			return
+		}
+		err = fmt.Errorf("list Access Points failed: %v", err)
+		return
+	}
+	for _, ap := range res.AccessPoints {
+		// check if AP exists with same Root Directory
+		if *ap.RootDirectory.Path == accessPointOpts.DirectoryPath {
+			// APs can be created by other than efs-csi-driver also. so check for pvcName Tag which was created by efs-csi-driver
```

Comment:
Do you mean check for tag so we know which ones are created by driver?

Reply:
done